### ISSUE TITLE
no-misused-generics: false positive with overload signature

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#438]
I've written the following code as part of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29806
The function needs to use generics to defer evaluation of the conditional type in return position.
```ts
type NullIfUndefined<T> = T extends undefined ? null : T;
type IfTrue<T, Then, Else> = T extends true ? Then : Else;

export function getItem<T extends boolean | undefined = false, U = undefined>(key: string, parse?: T, fallback?: U): IfTrue<typeof parse, unknown, string | NullIfUndefined<typeof fallback>>;
```

Currently the rule adds a failure for both `T` and `U` of `getItem`.

**Comment by ajafff** [comment]
Nevermind, it can be rewritten using the type parameter in the conditional type instead.
